test/FunctionsV2/OutOfProcTests.cs

```diff
@@ -371,5 +372,42 @@ public async Task InvokeLocalRpcEndpoint()
                 await host.StopAsync();
             }
         }
+
+        [Theory]
+        [InlineData("dotnet")]
+        [InlineData("DotNet")]
+        [InlineData("node")]
+        [InlineData("Node")]
+        [InlineData("python")]
+        [InlineData("Python")]
+        [InlineData("powershell")]
+        [InlineData("PowerShell")]
+        [InlineData("haskell")]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void WorkerRuntimeTypeFollowsSpec(string workerRuntime)
+        {
+            INameResolver nameResolver = new SimpleNameResolver(
+                new Dictionary<string, string>
+                {
+                    { "FUNCTIONS_WORKER_RUNTIME", workerRuntime },
+                });
+
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            var platformInformation = new DefaultPlatformInformation(nameResolver, loggerFactory);
+
+            var runtimeType = platformInformation.GetWorkerRuntimeType();
+
+            if (workerRuntime == "haskell")
```

Comment:
This makes this test fragile. I'm curious if we should just have a HashSet (initialize with StringComparison.OrdinalIgnoreCase) of the approved list of runtimes, and then make this check if the workerRuntime does not exist there.